src/protocols/secure_channel/CASESession.cpp

```diff
@@ -479,27 +471,26 @@ CHIP_ERROR CASESession::SendSigmaR2()
     SuccessOrExit(err);
 
     {
-        MutableByteSpan saltSpan(msg_salt.Get(), saltlen);
-        err = ConstructSaltSigmaR2(ByteSpan(msg_rand.Get(), kSigmaParamRandomNumberSize), mEphemeralKey.Pubkey(), ByteSpan(mIPK),
-                                   saltSpan);
+        MutableByteSpan saltSpan(msg_salt);
+        err = ConstructSaltSigmaR2(ByteSpan(msg_rand), mEphemeralKey.Pubkey(), ByteSpan(mIPK), saltSpan);
         SuccessOrExit(err);
-    }
 
-    err = mHKDF.HKDF_SHA256(mSharedSecret, mSharedSecret.Length(), msg_salt.Get(), saltlen, kKDFSR2Info, kKDFInfoLength, sr2k,
-                            kAEADKeySize);
-    SuccessOrExit(err);
+        HKDF_sha_crypto mHKDF;
+        err = mHKDF.HKDF_SHA256(mSharedSecret, mSharedSecret.Length(), saltSpan.data(), saltSpan.size(), kKDFSR2Info,
+                                kKDFInfoLength, sr2k, kAEADKeySize);
+        SuccessOrExit(err);
+    }
 
     // Construct Sigma2 TBS Data
-    msg_r2_signed_len = EstimateTLVStructOverhead(
-        static_cast<uint16_t>(mOpCredSet->GetDevOpCredLen(mTrustedRootId) + kP256_PublicKey_Length * 2), 3);
+    msg_r2_signed_len = EstimateTLVStructOverhead(mOpCredSet->GetDevOpCredLen(mTrustedRootId) + kP256_PublicKey_Length * 2, 3);
 
     VerifyOrExit(msg_R2_Signed.Alloc(msg_r2_signed_len), err = CHIP_ERROR_NO_MEMORY);
 
     {
         TLV::TLVWriter tlvWriter;
         TLV::TLVType outerContainerType = TLV::kTLVType_NotSpecified;
 
-        tlvWriter.Init(msg_R2_Signed.Get(), msg_r2_signed_len);
+        tlvWriter.Init(msg_R2_Signed.Get(), static_cast<uint32_t>(msg_r2_signed_len));
```

Comment:
Can we (in some other PR) also consider changing `CHIPTLV.h` to use `size_t` too?  Its use of uint32_t for lengths seems very intentional, but I just can't understand it.

Reply:
https://github.com/project-chip/connectedhomeip/pull/8951

Reply:
You raced me @bzbarsky-apple : https://github.com/project-chip/connectedhomeip/issues/8952